appium/webdriver/extensions/location.py

```diff
@@ -44,9 +44,10 @@ def set_location(self, latitude, longitude, altitude):
             "location": {
                 "latitude": float(latitude),
                 "longitude": float(longitude),
-                "altitude": float(altitude)
             }
         }
+        if altitude:
+            data['location']['altitude'] = float(altitude)
```

Comment:
please don't cast the values to float. the server code can properly handle both numbers and strings if they have parseable format